tests/services/test_criteria_matcher.py

```
@@ -33,30 +34,17 @@ def _mock_stat(
 
 
 @pytest.fixture
-def mock_file(tmp_path, mock_stat):
-    """Fixture to create a mock file with specific stat info."""
+def mock_file(tmp_path):
+    """Fixture to create a mock file. Does NOT mock stat itself."""
 
-    def _mock_file(filename="test.txt", **stat_kwargs):
+    @contextlib.contextmanager
+    def _mock_file(filename="test.txt", create_file=True):
```

Ruff (0.14.14)

[warning] Missing return type annotation for private function `_mock_file`

(ANN202)